pallets/did/src/utils.rs

```diff
@@ -0,0 +1,161 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+/// Verifies that an input string contains only URL-allowed ASCII characters.
+/// For more info about what those characters are, please visit the official RFC
+/// 3986.
+pub fn is_valid_ascii_url(input: &str) -> bool {
+	for c in input.chars() {
+		// Matches [0-9], [a-z], [A-Z], plus the symbols as in the RFC.
```

Comment:
You could probably also use `input.chars().find(|c| !match!(c, ...)).is_none()`.